From 21eb74155c9c5e2222ae67f770bc98993b40edc5 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 28 Aug 2018 11:15:34 +0800
Subject: [PATCH 4460/5242] MLK-19349 ARM64: dts: freescale: add i.mx8qxp
 ddr3l validation board dtb

commit  85208440ea54e0c3b0f825800f92f5bdde052e40 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds device tree support for i.MX8QXP DDR3L validation
board, it ONLY has 1GB memory, so CMA size needs to be reduced
when including the i.MX8QXP LPDDR4 ARM2 board device tree file.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
(cherry picked from commit 3005f91cef3057aa6eeb168415ec42e57fc993f9)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    1 +
 .../boot/dts/freescale/fsl-imx8qxp-ddr3l-val.dts   |   27 ++++++++++++++++++++
 2 files changed, 28 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-ddr3l-val.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 871f5c7..4c61fd5 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -77,6 +77,7 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-wm8962.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-dsi-rm67191.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-a0.dtb \
+				  fsl-imx8qxp-ddr3l-val.dtb \
 				  fsl-imx8dx-lpddr4-arm2.dtb \
 				  fsl-imx8dxp-lpddr4-arm2.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-ddr3l-val.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-ddr3l-val.dts
new file mode 100644
index 0000000..c9a39e8
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-ddr3l-val.dts
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qxp-lpddr4-arm2.dts"
+
+/ {
+	reserved-memory {
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0 0x14000000>;
+			alloc-ranges = <0 0x96000000 0 0x14000000>;
+			linux,cma-default;
+		};
+	};
+};
-- 
1.7.9.5

